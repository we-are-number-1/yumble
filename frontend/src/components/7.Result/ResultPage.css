.ResultTitle {
  -webkit-text-stroke: 4px black;
  color: white;
  font-family: Comic Sans MS;
  margin: 1vw 1vw 0vw;
  font-size: 50pt;
}

.MainContainer {
  display: flex;
  flex-direction: row;
  justify-content: space-evenly;
  align-items: center;
  width: 75vw;
  height: 100vh;
  border: 4px solid #000000;
  background-color: #f4cb9b;
  border-radius: 30px;
  margin: 5px;
}

.PieChart {
  width: 40vw;
  height: 60vh;
  display: flex;
  justify-content: center;
  align-items: center;
}

.DataVisual {
  width: 100vw;
  height: 50vh;
}

.ButtonReset {
  background: #F63D52;
  background-image: -webkit-linear-gradient(top, #F63D52, #9D1ED0);
  background-image: -moz-linear-gradient(top, #F63D52, #9D1ED0);
  background-image: -ms-linear-gradient(top, #F63D52, #9D1ED0);
  background-image: -o-linear-gradient(top, #F63D52, #9D1ED0);
  background-image: -webkit-gradient(to bottom, #F63D52, #9D1ED0);
  -webkit-border-radius: 20px;
  -moz-border-radius: 20px;
  border-radius: 20px;
  color: #FFFFFF;
  font-family: Brush Script MT;
  font-size: 40px;
  font-weight: 400;
  padding: 23px;
  -webkit-box-shadow: 1px 1px 20px 0 #000000;
  -moz-box-shadow: 1px 1px 20px 0 #000000;
  box-shadow: 1px 1px 20px 0 #000000;
  text-shadow: 1px 1px 20px #000000;
  border: solid #337FED 1px;
  text-decoration: none;
  display: inline-block;
  cursor: pointer;
  text-align: center;
  margin: 5%;
}

.ButtonReset:hover {
  border: solid #337FED 1px;
  background: #1E62D0;
  background-image: -webkit-linear-gradient(top, #1E62D0, #3D94F6);
  background-image: -moz-linear-gradient(top, #1E62D0, #3D94F6);
  background-image: -ms-linear-gradient(top, #1E62D0, #3D94F6);
  background-image: -o-linear-gradient(top, #1E62D0, #3D94F6);
  background-image: -webkit-gradient(to bottom, #1E62D0, #3D94F6);
  -webkit-border-radius: 20px;
  -moz-border-radius: 20px;
  border-radius: 20px;
  text-decoration: none;
}
