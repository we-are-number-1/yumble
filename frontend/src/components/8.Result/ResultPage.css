.ResultTitle {
    -webkit-text-stroke: 4px black;
    color: white;
    font-family: Comic Sans MS;
    margin: 1vw 1vw 0vw;
    font-size: 50pt;
  }
  
  .MainContainer {
    display: flex;
    flex-direction: row;
    justify-content: space-evenly;
    align-items: center;
    width: 75vw;
    height: 100vh;
    border: 4px solid #000000;
    background-color: #f4cb9b;
    border-radius: 30px;
    margin: 5px;
  }
  
  .PieChart {
    width: 40vw;
    height: 60vh;
    display: flex;
    justify-content: center;
    align-items: center;
  }

  .DataVisual {
    width: 100vw;
    height: 50vh;
  }