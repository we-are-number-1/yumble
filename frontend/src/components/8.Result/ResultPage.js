import React from 'react';

/**
 *
 * @return {*}
 */
function ResultPage() {
  return (
    <>
      <p>Result</p>
    </>
  );
}

export default ResultPage;
