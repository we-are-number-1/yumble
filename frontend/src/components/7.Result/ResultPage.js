import React, {useState, useEffect, useContext} from 'react';
import Help from '../Common/Help';
import '../Common/Help.css';
import MapModal from '../Common/MapModal';
import Icon from '../Common/MapsPinpoint';
import DataVisual from './DataVisual';
import SwipeCard from '../Common/SwipeCard';
import axios from 'axios';
import './ResultPage.css';
import {SocketContext} from '../../sockets/SocketContext';

// Dummy data, real data is retrived from sockets
const name = 'Lonestar';
const location = 'Botany';
const price = '$$$';
const rating = 4.0;

/**
 * @param {*} props {hasResult}
 * @return {*}
 */
function ResultPage(props) {
  const [ButtonPopup, setButtonPopup] = useState(false);
  const [MapPopup, setMapPopup] = useState(false);
  const [cardList, setCardList] = useState(null);
  const [data, setData] = useState({name, location, price, rating});
  const [pie, setPie] = useState(null);
  const [chart, setChart] = useState(false);
  const socketContext = useContext(SocketContext);
  // use this attribute to decide what component to render
  // null -> nothing
  // false -> no result decided
  // true -> result with pie chart
  // default is null because no props is passed in
  // need to have props due to the testing purpose
  const [hasResult, setHasResult] = useState(props.hasResult);

  useEffect(() => {
    document.title = 'Time to go eat!';
    axios
        .get('sessions/'+ socketContext.code)
        .then((res) => {
          setHasResult(false);
          setCardList(res.data.results.sort(
              function(a, b) {
                return b.numberOfVotes - a.numberOfVotes;
              }));
        })
        .catch(function(error) {
          console.log(error);
        });
  }, []);

  useEffect(()=>{
    if (cardList && cardList.length > 0) {
      setHasResult(true);
      const card = cardList[0];
      setData({
        name: card.name,
        location: card.location,
        price: card.price,
        rating: card.rating,
        images: card.images,
        coords: card.coords,
      });
      const pieChart = {};
      pieChart.labels = [];
      const votes = [];
      for (let i = 0; i < cardList.length; i++) {
        if (cardList[i].name && i < 6) {
          pieChart.labels.push(cardList[i].name);
          votes.push(cardList[i].numberOfVotes);
        } else {
          break;
        }
      }

      pieChart.datasets = [{
        label: '# of Votes',
        data: votes,
        borderColor: [
          'rgba(255, 99, 132, 0.2)',
          'rgba(54, 162, 235, 0.2)',
          'rgba(255, 206, 86, 0.2)',
          'rgba(75, 192, 192, 0.2)',
          'rgba(153, 102, 255, 0.2)',
          'rgba(255, 159, 64, 0.2)',
        ],
        backgroundColor: [
          'rgba(255, 99, 132, 1)',
          'rgba(54, 162, 235, 1)',
          'rgba(255, 206, 86, 1)',
          'rgba(75, 192, 192, 1)',
          'rgba(153, 102, 255, 1)',
          'rgba(255, 159, 64, 1)',
        ],
        borderWidth: 1,
      }];
      setPie(pieChart);
    }
  }
  , [cardList]);


  useEffect(() => {
    if (pie) {
      setChart(true);
    }
  }, [pie]);

  return (
    <>
      <div className='MakeCentre' id='ExtraHeight'>
        { hasResult === undefined ? <></> : hasResult === true ?
        <>
          <h1 className='ResultTitle'>Top Choice</h1>
          <div className='MainContainer'>
            <SwipeCard data={data}/>
            {chart&&<DataVisual className='DataVisual' data={pie}/>}
          </div>
          <button
            onClick={() => setMapPopup(true)}
            className='BigBtn'
            id='GoogleMaps_btn'
          >
            View on Google Maps
            <Icon />
          </button>
          <MapModal trigger={MapPopup} setTrigger={setMapPopup}
            restaurantLocation={data.coords} />
        </> : <>
          <h2 className='ResultTitle'> No Result Decided :( </h2>
          <h2 className='ResultTitle'> Wanna try another place ? </h2>
          <button className='ButtonReset' onClick={
            () => window.location.reload()}>
            Try Again :) </button>
        </>
        }
      </div>
      <button
        onClick={() => setButtonPopup(true)}
        className='SmallBtn'
        id='HelpButton'
      >help?</button>
      <Help trigger={ButtonPopup} setTrigger={setButtonPopup}>
        <p>You are all set! Click on [View on Google Maps]
          to see where you and your friends are going.</p>
      </Help>
    </>
  );
}

export default ResultPage;
