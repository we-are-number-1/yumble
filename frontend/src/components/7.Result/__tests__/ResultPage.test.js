import React from 'react';
import renderer from 'react-test-renderer';

import ResultPage from '../ResultPage';

test('ResultPage snapshot testing when there is result', () => {
  const tree = renderer.create(<ResultPage hasResult={true} />).toJSON();
  expect(tree).toMatchSnapshot();
});


