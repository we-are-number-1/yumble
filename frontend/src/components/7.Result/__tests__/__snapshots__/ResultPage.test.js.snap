// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultPage snapshot testing when there is result 1`] = `
Array [
  <h1
    className="Title"
  >
    yumble
  </h1>,
  <div
    className="container"
    style={
      Object {
        "marginTop": "4em",
        "maxHeight": "100%",
      }
    }
  >
    <div
      className="MakeCentre"
    >
      <div
        className="card"
        id="Card-field"
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "justifyContent": "center",
          }
        }
      >
        <h2
          className="text-center card-header"
          id="Card-Header"
          style={
            Object {
              "width": "100%",
            }
          }
        >
           
          Top Choice
        </h2>
        <div
          className="card-body"
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
              "justifyContent": "center",
            }
          }
        >
          <div
            className="justify-content-md-center row row-cols-lg-12"
            style={
              Object {
                "alignItems": "center",
                "display": "flex",
                "justifyContent": "center",
                "maxWidth": "95%",
              }
            }
          >
            <div
              className="col"
              style={
                Object {
                  "alignItems": "center",
                  "display": "flex",
                  "justifyContent": "center",
                  "marginTop": "1em",
                  "maxHeight": "50%",
                }
              }
            >
              <div
                className="SwipeCard "
                onClick={[Function]}
              >
                <div
                  className="CardImage"
                >
                  <img
                    className="CardImage"
                  />
                </div>
                <div>
                  <p
                    className="PlaceDetails"
                    id="name"
                  >
                     
                    Lonestar
                     
                  </p>
                </div>
                <p
                  className="PlaceDetails"
                  id="location"
                >
                   
                  Botany
                   
                </p>
                <p
                  className="PlaceDetails"
                  id="price"
                >
                   
                  $$$
                   
                </p>
                <p
                  className="PlaceDetails"
                  id="rating"
                >
                   
                  4.0
                   Rating
                </p>
              </div>
            </div>
            <div
              className="col-lg-12"
              style={
                Object {
                  "alignItems": "center",
                  "display": "flex",
                  "justifyContent": "center",
                }
              }
            >
              <button
                className="BigBtn btn btn-success"
                disabled={false}
                id="Resultbutton"
                onClick={[Function]}
                style={
                  Object {
                    "fontSize": "2em",
                    "marginBottom": "0.1em",
                    "marginTop": "0.1em",
                  }
                }
                type="button"
              >
                See results
              </button>
              <button
                className="BigBtn btn btn-info"
                disabled={false}
                onClick={[Function]}
                style={
                  Object {
                    "fontSize": "2em",
                    "marginBottom": "0.1em",
                    "marginLeft": "0.3em",
                    "marginTop": "0.1em",
                  }
                }
                type="button"
              >
                Restart
                 
              </button>
              
            </div>
            <div
              className="col-lg-12"
              style={
                Object {
                  "alignItems": "center",
                  "display": "flex",
                  "justifyContent": "center",
                }
              }
            >
              <button
                className="BigBtn btn btn-warning"
                disabled={false}
                id="GoogleMaps_btn"
                onClick={[Function]}
                style={
                  Object {
                    "marginBottom": "0.15em",
                    "marginTop": "0.15em",
                  }
                }
                type="button"
              >
                View on Google Maps
                <svg
                  height="30"
                  viewBox="-90 -90 600 600"
                  width="30"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M256 0C161.896 0 85.333 76.563 85.333 170.667c0 28.25 7.063 56.26 20.49 
 81.104L246.667 506.5c1.875 3.396 5.448 5.5 9.333 5.5s7.458-2.104 
 9.333-5.5l140.896-254.813c13.375-24.76 20.438-52.771 20.438-81.021C426.667
 76.563 350.104 0 256 0zm0 256c-47.052 0-85.333-38.281-85.333-85.333S208.948 
 85.334 256 85.334s85.333 38.281 85.333 85.333S303.052 256 256 256z"
                    fill="#ff6056"
                    height="100%"
                    width="100%"
                  />
                </svg>
              </button>
              
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  <button
    className="btn btn-info btn-lg"
    disabled={false}
    id="HelpButton"
    onClick={[Function]}
    type="button"
  >
    Help
  </button>,
  "",
]
`;
